### DISCUSSION TITLE
StatelessSession.insertAll(), updateAll(), deleteAll(), refreshAll()

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#5148] [Design Proposals]
Currently Hibernate classic has no way to batch operations in a `StatelessSession`. This problem was already solved in Hibernate Reactive by adding `insertAll()`, `updateAll()`, `deleteAll()`, `refreshAll()`.

Those operations are harder to implement in Hibernate classic because of how the whole batching infrastructure is done, but I still think it's well-worth the effort.

**Comment by Sanne** [comment]
Agreed, those would be useful.
